jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNCSARequestLog.java

```diff
@@ -489,4 +496,58 @@ public String getLogTimeZone()
     {
         return _logTimeZone;
     }
+    
+
+    private String extractUser(Request request)
```

Comment:
Well at least with the PR I have it would behave the same as it doing now no?
I do not set the authentication header (so it must have been set by someone else i.e. somewhere by Jetty). and then the logic does like it always has. So isn't this issue going to be there anyway?

Well maybe I do not understand the intention of the NCSA fully...

But I guess the question is: is an entry in the NCSA log a log of an authenticated user that has tried to access a resource, or is it an entry in the NCSA log a log of someone trying to access the log and he claims to be X but he has not been authenticated yet.

 In case the intention of the NCSA it is the former, well then the PR should most likely be rejected. In case we would like to know who actually did try to access the server (no matter if they where authenticated or not) then this is not possible at the moment. Hence the PR.

The issue I was having was that the authentication when you combine spring and jetty. It seems like logging is prior to authenticating (which is done by spring). In case the NCSA intention should only log authenticated user then the collaboration between spring and jetty is not setup correctly.